standalone-metastore/metastore-common/src/main/thrift/hive_metastore.thrift

```diff
@@ -2070,18 +2096,30 @@ struct CreateDatabaseRequest {
   8: optional string catalogName,
   9: optional i32 createTime,
   10: optional string managedLocationUri,
-  11: optional string type,
-  12: optional string dataConnectorName
+  11: optional DatabaseType type,
+  12: optional string dataConnectorName,
+  13: optional string remote_dbname
 }
 
 struct CreateDataConnectorRequest {
-  1: DataConnector connector
+  1: required DataConnector connector
 }
 
 struct GetDataConnectorRequest {
   1: required string connectorName
 }
 
+struct AlterDataConnectorRequest {
+  1: required string connectorName,
+  2: required DataConnector newConnector
+}
+
+struct DropDataConnectorRequest {
+  1: required string connectorName,
+  2: optional bool ifNotExists,
+  3: optional bool checkReferences
```

Comment:
Do we need this? This causes HMS to look at all the DBs that reference this connector name right? instead should we just define what should happen? We could allow the connector be dropped even with references (under the assumption that they want to re-create it differently, eg different properties like password, maybe secure the credentials etc) or do not allow dropping of the connector if there are references.

If we have this, I think clients can define the behavior or keep changing it on the long run, but I am worried about having to scan all DBs.

Reply:
Here the change is about creating a request-based API for the already existing API (which takes in all the above 3 parameters).
Currently, we don't use this flag. Yeah, your concern is genuine in the future when we implement this flag.